File: yyong/2025.05/250502_IF문_좀_대신_써줘.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const [N, M] = input[0].split(" ").map(Number);
const titles = [];

for (let i = 1; i <= N; i++) {
  const [name, power] = input[i].split(" ");
  titles.push([name, Number(power)]);
}

const result = [];

for (let i = N + 1; i < N + 1 + M; i++) {
  const num = Number(input[i]);

  let left = 0;
  let right = N - 1;

  while (left <= right) {
    const mid = Math.floor((left + right) / 2);
    const midPower = titles[mid][1];

    if (midPower < num) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }

  result.push(titles[left][0]);
}

console.log(result.join("\n"));

File: jun/2025.05/250502_IF문 좀 대신 써줘.py
"""
[BOJ] IF문 좀 대신 써줘 / 실버3
"""
import sys
input = sys.stdin.readline

n, m = map(int, input().split())

titles = []
limits = []

for _ in range(n):
    title, limit = input().split()
    titles.append(title)
    limits.append(int(limit))

powers = [int(input()) for _ in range(m)]

for power in powers:
    left = 0
    right = n - 1
    answer = 0

    while left <= right:
        mid = (left + right) // 2

        if power <= limits[mid]:
            answer = mid
            right = mid - 1

        else:
            left = mid + 1

    print(titles[answer])